exchange.go

```diff
@@ -442,7 +475,144 @@ func (re *RabbitExchangeImpl) Receive(exchange ExchangeSettings, queue QueueSett
 						closer()
 						for {
 							time.Sleep(time.Millisecond * 250)
-							channel, msgs, errChan, closer, err = getChannel()
+							channel, msgs, errChan, closer, err = re.getChannel(exchange, queue, retryExchangeName)
+							if err != nil {
+								log.Printf("Error connecting to rabbit %+v\n", err)
+							} else {
+								break
+							}
+						}
+					}
+
+				}
+			}
+		},
+		func() {
+			close(stop)
+		}, nil
+}
+
+func (re *RabbitExchangeImpl) BulkReceive(exchange ExchangeSettings, queue QueueSettings, maxWait int) (func(BulkMessageHandleFunc) error, func(), error) {
```

Comment:
```suggestion
func (re *RabbitExchangeImpl) BulkReceive(exchange ExchangeSettings, queue QueueSettings, maxWait time.Duration) (func(BulkMessageHandleFunc) error, func(), error) {
```
There is no need to through the type away here, just to add it back later. In its current form it is not clear if the unit is seconds or milliseconds. 
If we use time.Duration, the caller is clear about what to pass.